server/src/main/java/org/elasticsearch/cluster/routing/DelayedAllocationService.java

```diff
@@ -135,20 +135,20 @@ public DelayedAllocationService(ThreadPool threadPool, ClusterService clusterSer
         this.threadPool = threadPool;
         this.clusterService = clusterService;
         this.allocationService = allocationService;
-        clusterService.addListener(this);
```

Comment:
interesting... this change seems to introduce a legit bug that causes 
```
:server:integTest --tests "org.elasticsearch.cluster.allocation.ClusterRerouteIT.testDelayWithALargeAmountOfShards"
```

to fail. Will need to dive into this to understand why